src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionUpdatingService.cs

```diff
@@ -0,0 +1,75 @@
+using DonateHope.Core.DTOs.CampaignContributionDTOs;
+using DonateHope.Core.Errors;
+using DonateHope.Core.Mappers;
+using DonateHope.Core.ServiceContracts.CampaignContributionsServiceContracts;
+using DonateHope.Domain.Enums;
+using DonateHope.Domain.RepositoryContracts;
+using FluentResults;
+
+namespace DonateHope.Core.Services.CampaignContributionServices;
+
+public class CampaignContributionUpdatingService (
+    CampaignContributionMapper campaignContributionContributionMapper,
+    ICampaignContributionsRepository campaignContributionsRepository
+    ) : ICampaignContributionUpdatingService
+{
+    private readonly CampaignContributionMapper _campaignContributionMapper = campaignContributionContributionMapper;
+    private readonly ICampaignContributionsRepository _campaignContributionsRepository = campaignContributionsRepository;
+
+    public async Task<Result<CampaignContributionGetResponseDto>> UpdateCampaignContributionAsync(
+        CampaignContributionUpdateRequestDto updateRequestDto,
+        Guid userId
+    )
+    {
+        // Check if amount is greater than or equal to 0
+        if (updateRequestDto.Amount < 0)
```

Comment:
(Optional) Should we leverage FluentValidation which is already in the codebase for model validation? This allows services to focus more on the business logic. We can do it in the upcoming features